content/docs/connect/choose-connection.md

```diff
@@ -0,0 +1,163 @@
+
+## Connection types for typical applications
+
+- **E-commerce Platforms:** High concurrency and rapid read/write operations. Use pooled connections for handling high traffic and direct connections for inventory updates.
+- **Content Management Systems (CMS):** Frequent read operations and moderate write operations. Use direct connections for content updates and pooled connections for concurrent reads.
+- **Financial Applications:** High concurrency and secure transactions. Use direct connections for stability and security, and pooled connections for high-frequency reads.
+- **Analytics and Reporting Tools:** Long-running queries and high data throughput. Use direct connections for report generation and pooled connections for frequent data access.
+- **Serverless Applications:** Rapid scaling and short-lived connections. Use the serverless driver with HTTP for one-shot queries and WebSocket for persistent transactions.
+
+## Configuration Examples
+
+**Direct connections**:
+
+1. Install the Postgres client library (e.g., pg for Node.js).
+
+   ```bash
+   npm install pg dotenv
+   ```
+
+1. Set up the connection string in your application configuration.
+
+   ```javascript
+   const { Client } = require('pg');
+   const client = new Client({
+     connectionString: process.env.DATABASE_URL,
+   });
+   client.connect();
+   ```
+
+1. Use the client for database operations.
+
+   ```javascript
+   client.query('SELECT * FROM users', (err, res) => {
+     if (err) throw err;
+     console.log(res.rows);
+     client.end();
+   });
+   ```
+
+**Pooled Connections**:
+
+1. You don't need install PgBouncer; just use the `-pooler` suffix with your connection string.
+1. Modify the connection string to use this pooled endpoint.
+
+   ```javascript
+   const { Pool } = require('pg');
+   const pool = new Pool({
+     connectionString: process.env.DATABASE_URL_POOLER,
+   });
+   ```
+
+1. Use the pool for database operations.
+
+   ```javascript
+   pool.query('SELECT * FROM users', (err, res) => {
+     if (err) throw err;
+     console.log(res.rows);
+     pool.end();
+   });
+   ```
+
+**Serverless Driver**:
+
+1. Install the serverless driver.
+
+   ```bash
+   npm install @neondatabase/serverless
+   ```
+
+1. Configure the connection for HTTP or WebSocket.
+
+   ```javascript
+   const { Client } = require('@neondatabase/serverless');
+   const client = new Client({
+     connectionString: process.env.DATABASE_URL,
+   });
+   client.connect();
+   ```
+
+1. Use the client for database operations.
+
+   ```javascript
+   client.query('SELECT * FROM users', (err, res) => {
+     if (err) throw err;
+     console.log(res.rows);
+     client.end();
+   });
+   ```
+
+## Monitoring
```

Comment:
Maybe monitoring connections deserves its own topic.

Reply:
Moved to stash file for now